File: app/src/javax/java/com/adawolfs/mvp_java/mvp/BaseMvpView.java
package com.adawolfs.mvp_java.mvp;

import android.content.Context;
import android.support.annotation.StringRes;

/**
 * Created by adawolfs on 1/20/17.
 */

public interface BaseMvpView {

    Context getContext();

    void showError(String error);

    void showError(@StringRes int error);

    void showMessage(@StringRes int srtResId);

    void showMessage(String message);
}

File: app/src/kotlin/java/com/adawolfs/mvp/mvp/BaseMvpView.kt
package com.adawolfs.mvp.mvp

import android.content.Context
import android.support.annotation.StringRes

/**
 * Created by adawolfs on 1/20/17.
 */
interface BaseMvpView {

    fun getContext(): Context

    fun showError(error: String)

    fun showError(@StringRes error: Int)

    fun showMessage(@StringRes srtResId: Int)

    fun showMessage(message: String)
}